packages/resolvers/glob/src/GlobResolver.js

```diff
@@ -205,19 +215,27 @@ function set(obj, path, value) {
   obj[path[path.length - 1]] = value;
 }
 
-function generate(matches, isAsync, indent = '', count = 0) {
+function generate(
+  matches,
+  isAsync,
+  indent = '',
+  count = 0,
+  specifierType = 'commonjs',
+) {
   if (typeof matches === 'string') {
+    let key = `_temp${count++}`;
+    let imports;
     if (isAsync) {
-      return {
-        imports: '',
-        value: `() => import(${JSON.stringify(matches)})`,
-        count,
-      };
+      imports = `const ${key} = () => import(${JSON.stringify(matches)});`;
+    } else {
+      if (specifierType === 'esm') {
+        imports = `import ${key} from ${JSON.stringify(matches)};`;
```

Comment:
Should this be `import * as ${key} ...`? As it is, this will only return the default export, whereas the CommonJS and dynamic imports will return the full module namespace.

Reply:
@devongovett  So the problem is that importing with `import * as key` seems to trigger the JavaScript resolver to treat the import as requiring wrapping in an "esm in commonjs" compatibility layer (as I understand it)

The end result is you get this (pseudocode) as the export from the glob import:

```js 
{ 
  esmodule:true, 
  default: function(){ 
    return { 
      esmodule:true, 
      default: function()...
    }
  }
}
```

So each import is treated as requiring a wrapper inside the index which also gets a wrapper. This doesn't make sense to me because the outermost export _might_ need a compatibility wrapper if it were being built in a commonjs environment but the rest inside it shouldn't as they're being imported into an esm file.
Have I understood this correctly?

Reply:
I’m also not sure how to change this behaviour, I’m assuming it’s in the JavaScript packager, but not sure where and what impact that might have